exp/api/v1alpha3/azuremanagedcontrolplane_types.go

```diff
@@ -94,6 +98,23 @@ type AADProfile struct {
 	AdminGroupObjectIDs []string `json:"adminGroupObjectIDs"`
 }
 
+// APIServerAccessProfile - access profile for AKS API server.
+type APIServerAccessProfile struct {
```

Comment:
similar comment to https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1679/files#r706664083

I think you should add this for v1a4 only, and implement the conversion webhook? which should basically be a noop except storing the data to roundtrip v1a4 -> v1a3 -> v1a4 conversion

Reply:
Thanks @alexeldeib ! Reverted change in v1a3 and added one line in `azuremanagedcontrolplane_conversion.go` to restore the field during conversion.